### PR TITLE
EVA-3129: Initial prototype for genotype grid view

### PR DESCRIPTION
**Pull request by apriltuesday**

### CODE REVIEW

**Review by tcezard — COMMENTED** [review]
This seems to look good. Although I will have to pull it and check it more carefully. 
Have you attempted to write tests?

File: src/components/Pagination.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    render() {
+        const currentPage = this.props.currentPage;
+        const numPages = this.props.numPages;
+        // subset of page numbers to display
+        let pageNumbers = [...new Set([
+            1, 2, 3,
+            Math.max(currentPage-1, 1), currentPage, Math.min(currentPage+1, numPages),
+            numPages-2, numPages-1, numPages
+        ])].sort((a, b) => a - b);
```
Comment (tcezard) [inline src/components/Pagination.js]: This assume you'll have at least 3 pages. If you only have one page that won't work.

**Comment by apriltuesday** [comment]
I've not attempted tests yet... an entirely reasonable idea to add them though...

File: src/components/Pagination.js (hunk 1 above)
Comment (apriltuesday) [reply src/components/Pagination.js]: Apparently this is the hardest part of this project... I think I've fixed it though.

**Pull request merged**